app/components/edit/EditAddress.jsx

```diff
@@ -51,120 +65,140 @@ class EditAddress extends Component {
           state_province: '',
         };
       }
+      console.log("no address: " + newAddr)
       return { noLocation: !state.noLocation, address: newAddr };
     }, () => {
-      const { address } = this.state;
+      const { index, address } = this.state;
+      console.log("INDEX " + index)
       updateAddress(address);
     });
   }
 
   render() {
-    const { address = {} } = this.props;
+    const { index, item: address } = this.props;
+    const htmlID = `address${index}`;
     const { noLocation } = this.state;
     return (
-      <AddressForm
-        handleAddressChange={this.handleAddressChange}
-        handleNoLocationChange={this.handleNoLocationChange}
-        noLocation={noLocation}
-        address={address}
-      />
+      <div>
+        <label>
+          <input
+            type="checkbox"
```

Comment:
Moved this out of the `AddressForm` component because I don't want it to replicate with each child address, but it's not working as expected. It still re-renders every time you click the Add button.

Reply:
It looks like this is because anything that gets passed to the `editCollectionHOC` component gets rendered once per item. We'll have to split up the form into the part that should only render once and the part that should render once per item, and then make sure that the `editCollectionHOC` only gets called on the latter.